src/Rules/Symfony/TwigTemplateExistsRule.php

```diff
@@ -0,0 +1,104 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Symfony;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Symfony\TwigEnvironmentResolver;
+use PHPStan\Type\ObjectType;
+use function count;
+use function in_array;
+use function is_string;
+use function sprintf;
+
+/**
+ * @implements Rule<MethodCall>
+ */
+final class TwigTemplateExistsRule implements Rule
+{
+
+	/** @var TwigEnvironmentResolver */
+	private $twigEnvironmentResolver;
+
+	public function __construct(TwigEnvironmentResolver $twigEnvironmentResolver)
+	{
+		$this->twigEnvironmentResolver = $twigEnvironmentResolver;
+	}
+
+	public function getNodeType(): string
+	{
+		return MethodCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$templateArg = $this->getTwigTemplateArg($node, $scope);
+
+		if ($templateArg === null) {
+			return [];
+		}
+
+		$templateNames = [];
+
+		if ($templateArg->value instanceof Variable && is_string($templateArg->value->name)) {
+			$varType = $scope->getVariableType($templateArg->value->name);
+
+			foreach ($varType->getConstantStrings() as $constantString) {
+				$templateNames[] = $constantString->getValue();
+			}
+		} elseif ($templateArg->value instanceof String_) {
+			$templateNames[] = $templateArg->value->value;
+		}
```

Comment:
I think you can avoid the if/else and write
```suggestion
			$argType = $scope->getType($templateArg->value);

			foreach ($varType->getConstantStrings() as $constantString) {
				$templateNames[] = $constantString->getValue();
			}
```